src/main/java/org/opensearch/ad/ratelimit/RateLimitedRequestWorker.java

```diff
@@ -96,12 +96,44 @@ public void put(RequestType request) throws InterruptedException {
         public int size() {
             return this.content.size();
         }
+
+        public boolean isEmpty() {
+            return content.size() == 0;
+        }
+
+        /**
+         * Remove requests in the queue
+         * @param numberToRemove number of requests to remove
+         * @return removed requests
+         */
+        public int drain(int numberToRemove) {
+            int removed = 0;
+            while (removed <= numberToRemove) {
```

Comment:
Should removing an expired request count towards the drain total?

Reply:
Processing expiresed requests are done separately in clearExpiredRequests().  This is a method that are called just for removing requests' purpose.

Reply:
But is it possible that this drain function will clear only expired requests?

Reply:
It is possible. We can add a method like drainExpired for that purpose if required later.

Reply:
Is the purpose of the drain function to remove `numberToRemove` requests so that there are fewer requests in the queue that need to be executed? If so, because expired requests will never be executed, removing expired requests will not actually reduce the number of requests in the queue that need to be executed.

Reply:
yes.  The complication lies in numberToRemove that is calculated as (total requests - max requests) / # of queues.  I won't know how many of the numberToRemove are expired or not without iterating through all.  I want to avoid it as that is an unnecessary cost.  Expired requests will be cleared anyway. 

If I filter out expired requests, the removed unexpired requests will be uneven. Depending on the number of expired requests, we may remove more unexpired requests from one queue than another queue.

How about cleaning expired requests before calling the drain method?

Reply:
Ah I see. 

Yes cleaning expired requests before calling the drain method seems to make sense to me.

Reply:
cool, please check the new commits